ui/components/component-library/icon/icon.stories.js

```diff
@@ -207,6 +207,7 @@ export const Size = (args) => (
     <Icon {...args} size={SIZES.MD} />
     <Icon {...args} size={SIZES.LG} />
     <Icon {...args} size={SIZES.XL} />
+    <Icon {...args} size={SIZES.AUTO} />
   </Box>
 );
```

Comment:
Could this be a little more descriptive of it's use? (need to add a fragment tag at the top if you think this is worth doing)

```suggestion
    <Text
      as="span"
      variant={TEXT.BODY_SM}
      display={DISPLAY.FLEX}
      alignItems={ALIGN_ITEMS.CENTER}
    >
      <Icon {...args} size={SIZES.AUTO} marginRight={1} />
      <span>
        Size auto also exists and inherits the font-size of the parent element
      </span>
    </Text>
  </>
);
```

<img width="918" alt="Screen Shot 2022-09-28 at 12 26 36 PM" src="https://user-images.githubusercontent.com/8112138/192871317-8523979f-8d42-451d-95d6-4593ca2fecda.png">